lib/droplet_kit/resources/kubernetes_resource.rb

```diff
@@ -24,26 +24,39 @@ class KubernetesResource < ResourceKit::Resource
       action :delete, 'DELETE /v2/kubernetes/clusters/:cluster_id' do
       end
 
-      action :cluster_node_pools, 'GET /v2/kubernetes/clusters/:cluster_id/node_pools' do
+      action :cluster_node_pools, 'GET /v2/kubernetes/clusters/:id/node_pools' do
+        handler(200) { |response| KubernetesNodePoolMapping.extract_collection(response.body, :read) }
       end
 
-      action :cluster_find_node_pool, 'GET /v2/kubernetes/clusters/:cluster_id/node_pools/:pool_id' do
+      action :cluster_node_pool_find, 'GET /v2/kubernetes/clusters/:id/node_pools/:pool_id' do
+        handler(200) { |response| KubernetesNodePoolMapping.extract_single(response.body, :read) }
       end
 
-      action :cluster_node_pool_create, 'POST /v2/kubernetes/clusters/:cluster_id/node_pools' do
+      action :cluster_node_pool_create, 'POST /v2/kubernetes/clusters/:id/node_pools' do
+        body { |node_pool| KubernetesNodePoolMapping.representation_for(:create, node_pool) }
+        handler(201) { |response| KubernetesNodePoolMapping.extract_single(response.body, :read) }
+        handler(422) { |response| ErrorMapping.fail_with(FailedCreate, response.body) }
       end
 
-      action :cluster_node_pool_update, 'PUT /v2/kubernetes/clusters/:cluster_id/node_pools/:pool_id' do
+      action :cluster_node_pool_update, 'PUT /v2/kubernetes/clusters/:id/node_pools/:pool_id' do
+        body { |node_pool| KubernetesNodePoolMapping.representation_for(:update, node_pool) }
+        handler(202) { |response| KubernetesNodePoolMapping.extract_single(response.body, :read) }
+        handler(404) { |response| ErrorMapping.fail_with(FailedUpdate, response.body) }
       end
 
-      action :cluster_node_pool_delete, 'DELETE /v2/kubernetes/clusters/:cluster_id/node_pools/:pool_id' do
+      action :cluster_node_pool_delete, 'DELETE /v2/kubernetes/clusters/:id/node_pools/:pool_id' do
+        handler(202) { |response| true }
       end
 
-      action :cluster_node_pool_recycle, 'POST /v2/kubernetes/clusters/:cluster_id/node_pools/:pool_id/recycle' do
+      action :cluster_node_pool_recycle, 'POST /v2/kubernetes/clusters/:id/node_pools/:pool_id/recycle' do
```

Comment:
Following the previous comment, I think we should also drop `cluster` here, and re-arrange the format to be in `<verb>_<noun>` style, making this:
```
action :recycle_node_pool, ...
```